### PR TITLE
Repository, Artifact, Credentials

### PR DESCRIPTION
**Pull request by npetrangelo**
Added Repository class, and Artifact and Credentials records

### CODE REVIEW

File: src/main/java/dev/mccue/resolve/core/Artifact.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package dev.mccue.resolve.core;
+
+import java.util.Objects;
+
+public record Artifact(String url, boolean changing, boolean optional, Credentials credentials) {
+    @Override
+    public int hashCode() {
+        return 37 * (37 * (37 * (17 + url.hashCode()) + Boolean.hashCode(changing)) + Boolean.hashCode(optional)) + Objects.hashCode(credentials);
+    }
```
Comment (bowbahdoe) [inline src/main/java/dev/mccue/resolve/core/Artifact.java]: What is wrong with the default hashcode?

File: src/main/java/dev/mccue/resolve/core/Artifact.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        b.append(optional);
+        b.append(", changing = ");
+        b.append(changing);
+        if (credentials != null) {
+            b.append(", ");
+            b.append(credentials);
+        }
+        b.append(")");
+        return b.toString();
+    }
```
Comment (bowbahdoe) [inline src/main/java/dev/mccue/resolve/core/Artifact.java]: What is wrong with the default toString?

File: src/main/java/dev/mccue/resolve/core/Credentials.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package dev.mccue.resolve.core;
+
+import java.io.Serializable;
+
+public record Credentials(String user, String password) implements Serializable {
```
Comment (bowbahdoe) [inline src/main/java/dev/mccue/resolve/core/Credentials.java]: Why does this implement Serializable

File: src/main/java/dev/mccue/resolve/core/Credentials.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    @Override
+    public int hashCode() {
+        return 37 * (17 + user.hashCode()) + password.hashCode();
+    }
```
Comment (bowbahdoe) [inline src/main/java/dev/mccue/resolve/core/Credentials.java]: What is wrong with the default hashcode

File: src/main/java/dev/mccue/resolve/core/Credentials.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+    @Override
+    public String toString() {
+        return "Credentials(" + user + ", ****)";
+    }
```
Comment (bowbahdoe) [inline src/main/java/dev/mccue/resolve/core/Credentials.java]: This toString pattern is strange, normally records give the equivalent of

```java
"Credentials[user=" + user + ", password=" + password + "]";
```

I would expect this to be 

```java
"Credentials[user=" + user + ", password=***]";
```

if you were just altering it to redact the password.

File: src/test/java/dev/mccue/resolve/core/RepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package dev.mccue.resolve.core;
+
+import dev.mccue.resolve.maven.MavenRepository;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RepositoryTest {
+    @Test
+    public void testDefaults() {
+        assertInstanceOf(MavenRepository.class, Repository.defaults().get(0));
```
Comment (bowbahdoe) [inline src/test/java/dev/mccue/resolve/core/RepositoryTest.java]: What does this test meaningfully tell you? What does the first thing being a MavenRepository imply?

File: src/test/java/dev/mccue/resolve/core/RepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    }
+
+    @Test
+    public void testCentral() {
+        assertInstanceOf(MavenRepository.class, Repository.central());
```
Comment (bowbahdoe) [inline src/test/java/dev/mccue/resolve/core/RepositoryTest.java]: The return type is `MavenRepository`. This line doesn't seem meaningful

**Review by bowbahdoe — CHANGES_REQUESTED** [review]

File: src/main/java/dev/mccue/resolve/core/Credentials.java (hunk 3 above)
Comment (npetrangelo) [reply src/main/java/dev/mccue/resolve/core/Credentials.java]: It implemented Serializable in the Coursier java API, not sure why exactly

File: src/test/java/dev/mccue/resolve/core/RepositoryTest.java (hunk 6 above)
Comment (npetrangelo) [reply src/test/java/dev/mccue/resolve/core/RepositoryTest.java]: The important thing is that Maven central is a default. I'll fix that.

**Review by bowbahdoe — APPROVED** [review]

**Pull request merged**